### PR TITLE
IAP blog post

### PR DESCRIPTION
**Pull request by pntgoswami18**
Add IAP Blog Post

### CODE REVIEW

File: _posts/2020-09-25-programmatic-authentication-under-iap.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+   if (err) {
+       console.error("Error encountered", err);
+   } else {
+       console.info(`Headers added ${JSON.stringify(headersToSet)}, ${resp}`);
+   }
+});
+```
+
+This allowed us to add the authorization bearer token to every request that was being made through the Protractor browser instance.
+Additionally, we also had to set the `trustAllServers` parameter in browsermob-proxy to `true` to disable verification of all upstream servers' SSL certificates.
```
Comment (sushantwingified) [inline _posts/2020-09-25-programmatic-authentication-under-iap.md:150]: this line is extra information, can be removed.

File: _posts/2020-09-25-programmatic-authentication-under-iap.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+The IAP is definitely a good option to regulate access to private resources without having to implement a VPN tunneling. But throwing accessibility to automated agents in the mix created new interesting challenges here. There are some obvious pros to it:
+
+1. Easy access regulation to resources
+2. Fine grained control of access through users' and services' IAM roles
+3. Access through untrusted networks without the requirements of a VPN tunnel
+
+But there are some cons that we discovered along the way of implementing programmatic authentication:
+
+1. Slower execution speeds on private resources, a resultant of re-routing the access which is done to add authorization headers
+2. Added overhead of handling programmatic authentication in automation tests, which does not cater to the user flow being tested. Users won't be accessing the actual app over IAP.
```
Comment (sushantwingified) [inline _posts/2020-09-25-programmatic-authentication-under-iap.md:200]: usage of programmatic is redundant when talking about automation

File: _posts/2020-09-25-programmatic-authentication-under-iap.md (hunk 1 above)
Comment (pntgoswami18) [reply _posts/2020-09-25-programmatic-authentication-under-iap.md:147]: removed

File: _posts/2020-09-25-programmatic-authentication-under-iap.md (hunk 2 above)
Comment (pntgoswami18) [reply _posts/2020-09-25-programmatic-authentication-under-iap.md:196]: removed 'programmatic'

**Pull request merged**